src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemFilerTests.cs

```diff
@@ -105,6 +106,41 @@ public void WorkItemFiler_ValidatesSarifLogFileLocationArgument()
             action.Should().Throw<ArgumentNullException>();
         }
 
+        [Fact]
+        public void WorkItemFiler_SarifLogResultsShouldNotBeFiled()
+        {
+            SarifLog sarifLog = TestData.CreateSimpleLog();
+
+            // Set all results baseline state to 'Unchanged' so no work item should be filed.
+            sarifLog.Runs.SelectMany(run => run.Results).ForEach(result => result.BaselineState = BaselineState.Unchanged);
+
+            SarifWorkItemContext context = CreateAzureDevOpsTestContext();
+
+            context.SplittingStrategy = SplittingStrategy.None;
+
+            int numberOfResults = 0;
+            context.SetProperty(ExpectedWorkItemsCount, numberOfResults);
+            context.SetProperty(ExpectedFilingResult, FilingResult.None);
+
+            TestWorkItemFiler(sarifLog, context, true);
```

Comment:
>true [](http://example.com/codeflow?start=49&length=4)

Is it important for these tests to verify the adoClient == false code path?

I think probably it is important, because we want to ensure the GitHub filer also doesn't file work items?

Or is the relevant code sufficiently covered by the adoClient case?

Reply:
Parameter adoClient controls which FilingClient mock object it generates, adoClient or githubClient. Added/updated test cases also covers githubClient.